src/Symfony/Component/Messenger/Handler/RedispatchMessageHandler.php

```diff
@@ -22,8 +23,10 @@ public function __construct(
     ) {
     }
 
-    public function __invoke(RedispatchMessage $message): void
+    public function __invoke(RedispatchMessage $message): mixed
     {
-        $this->bus->dispatch($message->envelope, [new TransportNamesStamp($message->transportNames)]);
+        $envelope = $this->bus->dispatch($message->envelope, [new TransportNamesStamp($message->transportNames)]);
+
+        return $envelope->last(HandledStamp::class)?->getResult();
```

Comment:
Not sure how useful this is. This would only actually return a value if you redispatched to the sync transport, correct?

Reply:
That is correct, this is usefull when redispatching to a transport which has a configurable dsn. When a user configures the dsn of the transport to 'sync://' the result will still be returned. Mostly useful for development or a bundle where the transport is configured by the user. If you want I can also drop this